---
title: "Pull all workgroup biosamples"
author: "Genomics Center at Clalit"
date: "`r Sys.Date()`"
output: 
  html_document:
    theme: united
    toc: true
    toc_float: false
---

```{r initial_setup}
bs_analysis_folder<-'/media/SSD/Bioinformatics/Projects/basespace_automation_202206/'

```

```{r setup, include=FALSE,warning=F,message=F}

project_dir<-hpo_analysis_folder
knitr::opts_knit$set(root.dir=project_dir)
knitr::opts_chunk$set(echo = F)
library(ProjectTemplate)
setwd(project_dir)
load.project()

main_output_folder<-glue('./output/basespace_results_{Sys.Date()}')
#if (!dir.exists(main_output_folder)){dir.create(main_output_folder)}

```

```{get_all_biosamples}
workgroup <-'Genoox - collaborative'
# If you dont want to verify all samples but just update new samples, input old biosample_sheet
biosample_sheet_to_update<-'./data/biosamples_sheets/genoox_collaborative_2022-10-30.csv'

verify_workgroup(workgroup)
all_biosamples<-get_all_biosamples()
all_biosamples$project<-''
all_biosamples$project_id<-''
all_biosamples$has_fastq<-NA
n_biosamples<-nrow(all_biosamples)
bs_fqs_df<-list()
n_tested<-1
biosample_ids<-all_biosamples%>%pull(Id)

if (!is.na(biosample_sheet_to_update)){
  biosample_sheet_to_update<-readr::read_delim(biosample_sheet_to_update,delim='\t')
  all_biosamples<-all_biosamples%>%rows_update(biosample_sheet_to_update%>%select(Id,has_fastq))%>%mutate(has_fastq=as.numeric(has_fastq))
  # collect all biosamples that have no has_fastq value in the sheet to update and use only them
  biosample_ids<-all_biosamples%>%filter(is.na(has_fastq))%>%pull(Id)
}

for (biosample_id in biosample_ids){
  message(glue('Verifying {biosample_id}'))
  bs_fqs<-get_biosample_fastqs(biosample_id)
  biosample_attributes<-get_biosample_attributes(biosample_id)
  biosample_project<-biosample_attributes%>%pull(DefaultProject.Name)
  biosample_project_id<-biosample_attributes%>%pull(DefaultProject.Id)
  n_fastqs<-nrow(bs_fqs)
  print(n_fastqs)
  has_fastq<-ifelse(n_fastqs>0,1,0)
  all_biosamples<-all_biosamples%>%rows_update(tibble(Id=biosample_id,
                                                      project=biosample_project,
                                                      project_id=as.character(biosample_project_id),
                                                      has_fastq=has_fastq))
  bs_fqs_df[[as.character(biosample_id)]]<-bs_fqs
  message(glue('Reviewed {n_tested}/{n_biosamples} biosample ids'))
  n_tested<-n_tested+1
}

# write down all biosamples list
workgroup_name_for_file<-stringr::str_replace_all(tolower(workgroup),'[\\s\\-]+','_')
write.table(all_biosamples,sep='\t',file=glue('./data/biosamples_sheets/{workgroup_name_for_file}_{Sys.Date()}.csv'),
            row.names = F,
            quote = F)

```