---
title: "Run DRAGEN enrichment"
author: "Genomics Center at Clalit"
date: "`r Sys.Date()`"
output: 
  html_document:
    theme: united
    toc: true
    toc_float: false
---

```{r initial_setup}
bs_analysis_folder<-'/media/SSD/Bioinformatics/Projects/basespace_automation_202206/'

```

```{r setup, include=FALSE,warning=F,message=F}

project_dir<-hpo_analysis_folder
knitr::opts_knit$set(root.dir=project_dir)
knitr::opts_chunk$set(echo = F)
library(ProjectTemplate)
setwd(project_dir)
load.project()

main_output_folder<-glue('./output/basespace_results_{Sys.Date()}')
if (!dir.exists(main_output_folder)){dir.create(main_output_folder)}

```

```{r run_attributes}
# Set project name
app_name<-'DRAGEN Enrichment'
app_version<-'4.0.3'

workgroup <-'Sequencing Center'
verify_workgroup(workgroup)
run_sheet_name_prefix<-'popgen_wes_idt_20221027'
app_session_name<-'popgen_wes_idt_20221027'
ht_ref<-'hg38_altmaskedv2-cnv-hla-graph-anchored.v8'
target_bed_id<-'818709364'# in order to find the file id you need to get the appsession conent (in the right workgroup!) 
project_name<-'popgen_wes_idt_20221027'
project_name<-stringr::str_replace(project_name,'\\s+','_')

project_id<-get_project_id_by_name(project_name,create_project_if_not_available = T)

input_biosample_list_sheet<-readr::read_delim('./data/input_biosample_sheets/popgen_wes_idt_20221027.csv',delim = '\t')
input_list_sample_ids<-input_biosample_list_sheet%>%pull(BioSampleName)

generate_dragen_enrichment_runsheet(run_sheet_name_prefix = run_sheet_name_prefix,
                                    app_session_name = app_session_name,
                                    project_id = project_id,
                                    ht_ref = ht_ref,
                                    fixed_bed = 'custom',
                                    target_bed_id = target_bed_id,
                                    vc_target_bed_padding = 50,
                                    high_coverage = 1,
                                    lowpass = 1,
                                    input_list_sample_ids = input_list_sample_ids
                                    )

run_attributes<-parse_run_sheet(glue('./data/run_sheets/{run_sheet_name_prefix}.dragen_enrichment.csv'),attributes_list=list())
# now verify every bs has a fastq file
#run_dragen_enrichment(run_attributes)
launch_application(app_name = app_name,
                   app_version = app_version,
                   run_attributes = run_attributes)
### !!!! Please note, sometimes it takes alot of time for the run to start and the BS disconnects. it doesnt mean  the run wont go.. need to check whether it is still in the queued status !!!


```
