---
title: "Run DRAGEN germline"
author: "Genomics Center at Clalit"
date: "`r Sys.Date()`"
output: 
  html_document:
    theme: united
    toc: true
    toc_float: false
---

In order to run DRAGEN germline you need to prepare:
  - input biosample ids sheet - a sheet with the biosample IDs - in order to get it you can use the pull_workgroup_biosamples.Rmd script to produce a file with ALL the biosamples (and whether they have fastq files). You can then easily select the sample you want to run and make a new sample sheet. 

  - project name - if it doesnt exist, it will be generated
  - max number of samples - since running too many samples together causes BS CLI it to crash

```{r initial_setup}
bs_analysis_folder<-'/media/SSD/Bioinformatics/Projects/basespace_automation_202206/'

```

```{r setup, include=FALSE,warning=F,message=F}

project_dir<-bs_analysis_folder
knitr::opts_knit$set(root.dir=project_dir)
knitr::opts_chunk$set(echo = F)
library(ProjectTemplate)
setwd(project_dir)
load.project()

main_output_folder<-glue('./output/basespace_results_{Sys.Date()}')
#if (!dir.exists(main_output_folder)){dir.create(main_output_folder)}

```

# Run Attributes

```{r run_attributes}
# Set project name
app_name<-'DRAGEN Germline'
app_version<-'4.0.3'
max_num_of_samples_per_run<-5

workgroup <-'Sequencing Center'
verify_workgroup(workgroup)
run_sheet_name_prefix<-'harvard_wgs_idt_hg19_20221113'
app_session_name<-run_sheet_name_prefix
ht_ref<-'hg19_altmaskedv2-cnv-hla-graph-anchored.v8'

project_name<-run_sheet_name_prefix
project_name<-stringr::str_replace(project_name,'\\s+','_')

project_id<-get_project_id_by_name(project_name,create_project_if_not_available = T)
```

# Option 1: Run DRAGEN Germline using an input sample sheet

```{r run_biosample_list}
input_biosample_list_sheet<-readr::read_delim('./data/input_biosample_sheets/harvard_wgs_idt_hg19_20221113.csv',delim = '\t')
input_list_sample_ids<-input_biosample_list_sheet%>%pull(BioSampleName)

sample_chunks<-split(input_list_sample_ids,1:max_num_of_samples_per_run)

chunk_num<-1
for (chunk in sample_chunks){
  chunk_sample_ids<-chunk
  message(glue('Running chunk {chunk_num} with: {paste0(chunk_sample_ids,collapse=",")}'))
  chunk_run_sheet_prefix<-glue('{run_sheet_name_prefix}_{chunk_num}')
  generate_dragen_germline_runsheet(run_sheet_name_prefix = chunk_run_sheet_prefix,
                                    app_session_name = app_session_name,
                                    project_id = project_id,
                                    ht_ref = ht_ref,
                                    input_list_sample_ids = chunk_sample_ids)
  run_attributes<-parse_run_sheet(glue('./data/run_sheets/{chunk_run_sheet_prefix}.dragen_germline.csv'),attributes_list=list())
  # now verify every bs has a fastq file
  #run_dragen_enrichment(run_attributes)
  launch_application(app_name = app_name,
                     app_version = app_version,
                     run_attributes = run_attributes)
  chunk_num<-chunk_num+1
}

### !!!! Please note, sometimes it takes alot of time for the run to start and the BS disconnects. it doesnt mean  the run wont go.. need to check whether it is still in the queued status !!!
```

# Option 2: Run DRAGEN Enrichment using a run-sheet

```{r run_runsheet}
run_sheet<-'bgu_wes_idt_run_019_021_9.dragen_enrichment.csv'
run_attributes<-parse_run_sheet(glue('./data/run_sheets/{run_sheet}'),attributes_list=list())
  # now verify every bs has a fastq file
  #run_dragen_enrichment(run_attributes)
launch_application(app_name = app_name,
                   app_version = app_version,
                   run_attributes = run_attributes)

```
